service/awsconfig/v5/resource/loadbalancer/current.go

```diff
@@ -2,9 +2,85 @@ package loadbalancer
 
 import (
 	"context"
+
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/awsconfig/v5/key"
+)
+
+const (
+	cloudProviderClusterTagValue = "owned"
+	cloudProviderServiceTagKey   = "kubernetes.io/service-name"
 )
 
-// TODO
 func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
-	return nil, nil
+	currentState := LoadBalancerState{}
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return currentState, microerror.Mask(err)
+	}
+
+	currentState, err = r.clusterLoadBalancers(customObject)
+	if err != nil {
+		return currentState, microerror.Mask(err)
+	}
+
+	return currentState, nil
+}
+
+func (r *Resource) clusterLoadBalancers(customObject v1alpha1.AWSConfig) (LoadBalancerState, error) {
+	lbState := LoadBalancerState{}
+
+	// We get all load balancers because the API does not allow tag filters.
+	output, err := r.clients.ELB.DescribeLoadBalancers(&elb.DescribeLoadBalancersInput{})
+	if err != nil {
+		return lbState, microerror.Mask(err)
+	}
+
+	allLBNames := []*string{}
+	for _, lb := range output.LoadBalancerDescriptions {
+		allLBNames = append(allLBNames, lb.LoadBalancerName)
+	}
+
+	tagsInput := &elb.DescribeTagsInput{
+		LoadBalancerNames: allLBNames,
+	}
+	tagsOutput, err := r.clients.ELB.DescribeTags(tagsInput)
```

Comment:
I would appreciate a comment describing why we need to do this here because this is AWS magic for me I do not understand. The relationship between LBs and their tags is not quite clear to me. I would have thought we have the LBs already so why do we need the tags now?

Reply:
As I understand is to filter them in the loop below for ones containing cluster tag & service tag.

Reply:
Yes we have to get the tags via another API call and then filter. I've added more comments.